onvif/src/discovery/mod.rs

```diff
@@ -115,9 +117,20 @@ pub async fn discover(duration: Duration) -> Result<impl Stream<Item = Device>,
     let xml_receiver = ReceiverStream::new(xml_receiver);
     let device_receiver = ReceiverStream::new(device_receiver);
 
+    let mut known_responses = HashSet::new();
```

Comment:
> are the XMLs big?

I think around 1 KB, but who knows what the network device can send to us :smile: 

> memory usage is only increased for a brief time

It's for the full duration of discovery because we don't know when the duplicate arrives.

> tiny possibility of a hash collision

Even in case of collision, it's not that bad because the user can request discovery again, and it'll work the second time because there's randomness in the data.